tests/testPPMb3D.py

```diff
@@ -0,0 +1,228 @@
+            reference_catalog=self.reference_catalog[:, :3],
+            log=self.log)
+
+        matches, distances, pattern_idx, shift = self.pyPPMb.match(
+            source_catalog=self.source_catalog, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(matches), len(self.reference_catalog))
+        self.assertTrue(np.all(distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testOptimisticMatch(self):
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_catalog[:, :3],
+            log=self.log)
+
+        matches, distances, pattern_idx, shift = self.pyPPMb.match_optimistic(
+            source_catalog=self.source_catalog, n_check=9, n_match=6,
+            max_n_patterns=100, max_shift=60., max_rotation=6.0, max_dist=5.,
+            min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(matches), len(self.reference_catalog))
+        self.assertTrue(np.all(distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testMatchSkip(self):
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_catalog[:, :3],
+            log=self.log)
+
+        matches, distances, pattern_idx, shift = self.pyPPMb.match(
+            source_catalog=self.source_catalog, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=np.array([0]))
+        self.assertEqual(len(matches), len(self.reference_catalog))
+        self.assertTrue(np.all(distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testMatchMoreSources(self):
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_catalog[:500, :3],
+            log=self.log)
+
+        matches, distances, pattern_idx, shift = self.pyPPMb.match(
+            source_catalog=self.source_catalog, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60.0, max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(matches), len(self.reference_catalog[:500]))
+        self.assertTrue(np.all(distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testMatchMoreReferences(self):
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_catalog[:, :3],
+            log=self.log)
+
+        matches, distances, pattern_idx, shift = self.pyPPMb.match(
+            source_catalog=self.source_catalog[:500], n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=1.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(matches), len(self.reference_catalog[:500]))
+        self.assertTrue(np.all(distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testShift(self):
```

Comment:
This rotation of the unit sphere will appear as a rotation on our "focal plane". Added text to the doc string explaining this.